sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ConnectionOptions.java

```diff
@@ -24,10 +24,11 @@
     private final Scheduler scheduler;
     private final String fullyQualifiedNamespace;
     private final CbsAuthorizationType authorizationType;
+    private final String applicationId;
 
     public ConnectionOptions(String fullyQualifiedNamespace, TokenCredential tokenCredential,
             CbsAuthorizationType authorizationType, AmqpTransportType transport, AmqpRetryOptions retryOptions,
-            ProxyOptions proxyOptions, Scheduler scheduler) {
+            ProxyOptions proxyOptions, Scheduler scheduler, String applicationId) {
```

Comment:
It's better to pass the whole ClientOptions here since we might need the headers and other options that will be added there. It's easier to plumb this in once and not keep adding new params to this constructor.